lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -34,10 +37,20 @@ def to_native(self, course_overview):
                 kwargs={'course_id': course_id},
                 request=request
             )
+            try:
+                get_discussion_course_or_404(CourseKey.from_string(course_id), request.user)
```

Comment:
@wajeeha-khalid very good question!  The course_overview portion itself was a very big change in our system.  So it made sense to make an incremental change.  (We have a lot of backward compatibility issues to worry about and legacy courses and code.)  Yes, we are moving towards exactly what you are saying.  I, in fact, had the same thought earlier this week.  :)